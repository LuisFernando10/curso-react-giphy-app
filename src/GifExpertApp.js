
    import React, { Fragment, useState } from 'react';
    import { CategoryAdd } from './components/CategoryAdd';
    import { GifGrid } from './components/GifGrid';
    //import PropTypes from 'prop-types'

    const GifExpertApp = () => {

        //const categories = [ 'One Punch', 'Samurai X', 'Dragon Ball' ];
        const [ categories, setCategories ] = useState([ 'One Punch' ]);

        /*const handleAdd = () => {
            
            setCategories( [ ...categories, 'HunterXHunter' ] );
            //setCategories( category => [ ...categories, 'HunterXHunter' ] );
        };*/

        return(
            <>
                <h2>GifExpertAps</h2>
                <CategoryAdd setCategories={ setCategories }/>
                <hr/>
                <ol>
                    {
                        categories.map( category => (
                            <GifGrid
                                key={ category }
                                category={ category }
                            />
                        ))
                    }
                </ol>
            </>
        );
    };

    export default GifExpertApp;
    
    