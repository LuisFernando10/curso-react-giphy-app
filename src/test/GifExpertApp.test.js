
    import GifExpertApp from '../GifExpertApp';
    import { shallow } from 'enzyme'

    describe('Prueba en: GifExpertApp.js', () => { 
        
        const wrapper = shallow( <GifExpertApp /> );

        test('Debe: Retornar componente <GifExpertApp />', () => { 
            expect( wrapper ).toMatchSnapshot();    
        });

        test('Debe: Mostrar listado de categorÃ­as', () => { 
        
            const categories = ['Yuyu Hakusho', 'Death Note'];
            const wrapper = shallow( <GifExpertApp defaultCategory={ categories } /> );

            expect( wrapper ).toMatchSnapshot();
            expect( wrapper.find('GifGrid').length ).toBe( categories.length )
        })
     })