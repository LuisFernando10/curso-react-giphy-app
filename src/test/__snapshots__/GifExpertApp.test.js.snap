// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en: GifExpertApp.js Debe: Mostrar listado de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertAps
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Yuyu Hakusho"
      key="Yuyu Hakusho"
    />
    <GifGrid
      category="Death Note"
      key="Death Note"
    />
  </ol>
</Fragment>
`;

exports[`Prueba en: GifExpertApp.js Debe: Retornar componente <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertAps
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
